### PR TITLE
BUG: Fix for json lines issue with backslashed quotes

### PR DESCRIPTION
**Pull request by joshowen**
- [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

This is an additional fix to:
https://github.com/pandas-dev/pandas/pull/14429
https://github.com/pandas-dev/pandas/pull/14391

### CODE REVIEW

File: doc/source/whatsnew/v0.19.2.txt [hunk 1]
```diff
@@ -59,7 +59,7 @@ Bug Fixes
 - Bug in clipboard functions on Windows 10 and python 3 (:issue:`14362`, :issue:`12807`)
 - Bug in ``.to_clipboard()`` and Excel compat (:issue:`12529`)
 
-
+- Bug in to_json with lines=true containing backslashed quotes (:issue:`14693`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.2.txt]: ``.to_json()`` with ``lines=True`` specified, containing .....

**Comment by joshowen** [comment]
This fixed an edge case, but ultimately there are a bunch more.  I'm moving away from using this to the following code:
```
        from io import StringIO
        import jsonlines
        with StringIO() as buf:
            writer = jsonlines.Writer(buf)
            writer.write_all(ujson.loads(s))
            writer.close()
            buf.seek(0)
            return buf.read()
```

Their solution is fairly simple, but I'm not comfortable enough to update the vendored ujson package.
https://github.com/wbolster/jsonlines/blob/master/jsonlines/jsonlines.py

I've added the following PR to speed it up a bit:
https://github.com/wbolster/jsonlines/pull/24

**Comment by joshowen** [comment]
@jreback I think the right way to do this is to use jsonlines or to build its functionality into ujson rather than trying to transform the json formatted output.  What do you think?

**Comment by wbolster** [comment]
jfyi, `StringIO.getvalue()` is typically used to obtain the value as a a string, instead of `.seek(0)` and `.read()`.

**Comment by wbolster** [comment]
also, there is no need to use `StringIO` as a context manager, while it does make sense to use `jsonlines.Writer` as a context manager:

``` python
import io
import jsonlines

buf = io.StringIO()
with jsonlines.Writer(buf) as writer:
    writer.write_all(ujson.loads(s))
return buf.getvalue()
```

**Comment by jreback** [comment]
@joshowen 

> @jreback I think the right way to do this is to use jsonlines or to build its functionality into ujson rather than trying to transform the json formatted output. What do you think?

IIRC from the original issue, @aterrel and I had discussed this. Though its pretty performant now, the correct approach is to put it in the custom ujson code that pandas uses. That is somewhat more involved (though probably pretty straightforward).

**Pull request closed without merging**